docs/source/migrate/index.rst

```diff
@@ -0,0 +1,274 @@
+
+.. code-block:: daml
+
+  daml 1.2
+  module Foo where
+
+  template Foo
+    with
+      a : Int
+      b : Text
+      p : Party
+    where
+      signatory p
+
+
+Here is typical error message in this case:
+
+.. code-block:: shell
+
+
+  daml/Foo.daml:22:30: error:
+  • Could not deduce (DA.Upgrade.Conv
+  (DA.Generics.M1
+  DA.Generics.S
+  ('DA.Generics.MetaSel
+  ('DA.Generics.MetaSel0
+  ('Some "p")
+  'DA.Generics.NoSourceUnpackedness
+  'DA.Generics.NoSourceStrictness))
+  (DA.Generics.Rec0 Party))
+  (DA.Generics.S1
+  ('DA.Generics.MetaSel
+  ...
+
+The important hint is that the compiler is not able to deduce that our data type is an instance of
+the ``DA.Upgrade.Conv`` class and hence not convertible. In this case you will have to add your own
+upgrade/rollback templates to ``daml/Foo.daml``, that describe how to convert a contract of the
+template in question of package ``foo-1.0.0`` to one of ``foo-2.0.0`` and vice versa. For example
+
+.. code-block:: daml
+
+   template FooUpgrade
+    with
+        op : Party
+    where
+        signatory op
+        nonconsuming choice Upgrade: ContractId B.Foo
+            with
+                inC : ContractId A.Foo
+                sigs : [Party]
+            controller sigs
+                do
+                    d <- fetch inC
+                    assert $ fromList sigs == fromList (signatory d)
+                    create $ Foo with a = d.a; b = "updated"; p = d.p
+
+   template FooRollback
+    with
+        op : Party
+    where
+        signatory op
+        nonconsuming choice Rollback: ContractId A.Foo
+            with
+                inC : ContractId B.Foo
+                sigs : [Party]
+            controller sigs
+                do
+                    d <- fetch inC
+                    assert $ fromList sigs == fromList (signatory d)
+                    create $ Foo with a = d.a; p = d.p
+
+In a future iteration, we will also be able to automatically convert data types for which merely
+embeddings/projections exists instead of full isomorphisms and the above example would compile
+without further manual help.
+
+Deploying the migration
+-----------------------
+
+Once you've succeeded building the ``foo-upgrade-2.0.0`` package you can deploy it on the ledger
+together with the ``foo-2.0.0`` package. Optionally you can bundle it with the ``foo-2.0.0`` package
+into a single DAML archive by running
+
+.. code-block:: shell
+
+  daml damlc merge-dars foo-2.0.0/.daml/dist/foo-2.0.0.dar foo-upgrade-2.0.0/.daml/dist/foo-upgrade-2.0.0.dar --package-name foo-2.0.0-with-upgrades.dar
+
+You find more information on how to deploy DAML archive packages :ref:`here <deploy-ref_index>` .
+After the ``foo-upgrade-2.0.0`` package has been deployed on the ledger, there exists for every
+contract defined in ``foo-1.0.0`` a DAML workflow with a choice to upgrade it to ``foo-2.0.0`` or
+roll it back. A future iteration might add DAML triggers to carry out the upgrade process
```

Comment:
Let's please remove the last sentence. I can be misread as a promise.

Reply:
ok